src/util/simplify_expr.cpp

```diff
@@ -1429,11 +1428,9 @@ bool simplify_exprt::get_values(
   return true;
 }
 
-bool simplify_exprt::simplify_lambda(exprt &)
+simplify_exprt::resultt<> simplify_exprt::simplify_lambda(const exprt &expr)
```

Comment:
If we're touching this interface, can we make it consume a `const array_comprehension_exprt &expr` please?

Reply:
Let's defer -- there is the very independent issue of the confusion between array comprehension and lambda expressions (as in Chruch's lambda calculus). Look at 5f0808c59cad3f8e3c6fbffac21.